core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -783,6 +783,24 @@ package object config {
       .booleanConf
       .createWithDefault(false)
 
+  private[spark] val CACHE_CHECKPOINT_PREFERRED_LOCS =
+    ConfigBuilder("spark.rdd.checkpoint.cachePreferredLocs")
+      .internal()
+      .doc("Whether to cache preferred locations of checkpointed RDD. Caching preferred " +
+        "locations can relieve query loading to DFS and save the query time. The drawback " +
+        "is that the cached locations can be possibly outdated and lose data locality. " +
+        "This location cache expires by `spark.rdd.checkpoint.cachePreferredLocsExpireTime` " +
+        "minutes.")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val CACHE_CHECKPOINT_PREFERRED_LOCS_EXPIRE_TIME =
```

Comment:
Can we have one optional config instead of two? like, if the time isn't specified or is 0, don't cache?

Reply:
This is good idea. We can just have one config, instead of two.